File: Presentazione/Presentazione/DTO/TestoDTO.cs
using Presentazione.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Presentazione.DTO
{
    public class TestoDTO
    {
        public long id { get; set; }

        public string text { get; set; }

        public string dimensione { get; set; }

        public string posizione { get; set; }

        public int fontSize { get; set; }

        public string colore { get; set; }

        public int fontStyle { get; set; }

        public DiapositivaDTO diapositiva { get; set; }
    }
}

File: src/dto/testo.dto.ts
import { fontStyleText } from './fontStyleText.enum';
import { Colore } from './colore.obj';
import { DiapositivaDTO } from './diapositiva.dto';


export class TestoDTO{
  id: number;
  text: string;
  dimensione: string;
  posizione: string;
  fontSize: number;
  colore: Colore;
  fontStyle: number;
  diapositiva: DiapositivaDTO;

  constructor(diapositiva: DiapositivaDTO,id?: number, text?: string, dimensione?: string, posizione?: string,
             fontSize?: number, colore?: Colore, fontStyle?: fontStyleText){
        this.id = id ? id : 0;
        this.text = text ? text : "Nuovo Testo";
        this.dimensione = dimensione ? dimensione: "50";
        this.posizione = posizione? posizione : "1_1";
        this.fontSize = fontSize ? fontSize: 20;
        this.colore = colore ? colore: new Colore(255,255,255,100);
        this.fontStyle = fontStyle ? fontStyle:0;
        this.diapositiva = diapositiva;

  }
}
